### DISCUSSION TITLE
Is it possible to create a database from a raw APK file?

### DISCUSSION THREAD

**Discussion by jtjisgod** [github/codeql#10168] [General]
I want to create a database from a raw APK file.
But I have no idea how to create a database.

**Comment by aibaars** [comment]
CodeQL can only create database from source files. 

It might be possible to decompile the class files in the APK into source code and make a database from those. However, this is likely to be quite a hassle and the resulting database won't be great because decompiled code tends to be quite hard to understand.

**Reply by jtjisgod** [reply]
Thank you for reply, Aibaars.

I considered decompiling, too.

But I know creating database with compiled language is need to build for extracting relationship.

Is it possible to create a database without building?

**Reply by aibaars** [reply]
Yes, to create a database for Java, CodeQL needs a build command that compiles all the source files. This command could simply be a large `javac` command line compiling all the source files together in one go. 

It's not really possible to create a database without building. You could attempt to invoke the Java extractor directly, but this does not make things any easier. The Java extractor is basically a modified `javac`, so getting it to run is just as hard (perhaps even a little harder)  than the normal `javac`.
